### ISSUE TITLE
Weird JSON compare error

### ISSUE THREAD

**Issue by justmara** [apiaryio/dredd#902]
For some reason json response validation fails because of characters that does not actually exist in api response. On the image below red underlines are strings shown in 'Diff' report, and green ones - from the 'Real' tab.
![dredd-json](https://user-images.githubusercontent.com/4157587/31936885-8eec27e6-b8bb-11e7-845c-7b26629c7610.png)
In reality there is clean json that passes any validation, but dredd parsing inserts some weird characters to it.

**Comment by honzajavorek** [comment]
Thanks for the report! I assume the screenshots are from the Apiary reporter. I see some Cyrillic in the JSON, so my first guess would be there are encoding issues. But let's try to debug this properly 🙂 

Would you be able to post here a simple (as simple as possible) API Blueprint or Swagger, which consistently reproduces the problem?

One more question - does this cause the actual validation to fail? Does Dredd testing pass with/without the `--reporter=apiary`?

**Comment by justmara** [comment]
@honzajavorek yes, this definitely looks like encoding issue ;) Also there is related (I think so) validation bug, that makes me think about encoding:

> Header 'content-type' has value 'application/json; charset=utf-8' instead of 'application/json'

And yes, dredd still fails even without reporting to apiary. It doesn't say whats wrong in console, simply fails.

**Comment by honzajavorek** [comment]
Could you please provide the sample description so I could reproduce it?

BTW, by specification JSON is always UTF-8, so the charset is redundant, although many applications and frameworks add it by default. Regardless of that, it is definitely treated as UTF-8 in most places and definitely in all places in Dredd.

> Header 'content-type' has value 'application/json; charset=utf-8' instead of 'application/json'

This is a valid error in case your backend uses `application/json; charset=utf-8`, but you have `Content-Type` specified as `application/json` in your API description and not as `application/json; charset=utf-8`.

**Comment by justmara** [comment]
How could I make a sample, when it fails validating over our real api response? Not over apiary "response" model from apib file.

**Comment by honzajavorek** [comment]
Ah! I'm sorry, I didn't get this bit from the previous discussion. Hmmm 🤔 Would you be able to fork & modify the [dredd-example](https://github.com/apiaryio/dredd-example/)'s server implementation in `app.js` so it fails in a similar way? E.g. [the first response](https://github.com/apiaryio/dredd-example/blob/master/app.js#L30:L38).
